packages/app/components/UsersTabContent.tsx

```diff
@@ -0,0 +1,188 @@
+        contractAddressesByNetworkId[chainId]?.passportFactory || "";
+        return contract;
+    }, [web3, chainId]);
+
+    useEffect(() => {
+        // Get all memberships assosciated with wallet
+        if (contractInstance.options.address) {
+            setShowLoading(true);
+            (contractInstance.methods.getMemberships() as ContractSendMethod)
+            .call({
+                from: address,
+            })
+            .then((r: string[]) => {setMAddresses(r)})
+            .catch(console.error)
+            .finally(() => setShowLoading(false));
+        }
+    }, [contractInstance, address, setMAddresses]);
+
+    useEffect(() => {
+        if(selectedOption){
+            setShowLoading(true);
+            const contract = new web3.eth.Contract(getAbiFromJson(passportJson));
+            contract.options.address = selectedOption;
+            // Get tokenIds which have been bought
+            contract.getPastEvents('Transfer', {
+                filter: {
+                    _from: '0x0000000000000000000000000000000000000000'
+                },
+                fromBlock: 0
+            }).then((events) => {
+                const tokenIds = events.map((event) => event.returnValues.tokenId)
+                return tokenIds;
+            }).then((tokenIds) => {
+                // Get the owner of each of these bought tokens
+                const ownerPromises = tokenIds.map((tokenId) => {
+                    return contract.methods.ownerOf(tokenId).call();
+                });
+                Promise.all(ownerPromises).then((ownerAddrs) => {
+                    const owners: {[key: string]: number} = {};
+                    ownerAddrs.forEach((ownerAddr) => {
+                        owners[ownerAddr] = (owners[ownerAddr]+1) || 1;
+                    });
+                    setUsers(owners);
+                })
+                .finally(() => setShowLoading(false));
+            });
+        }
+    }, [selectedOption, setUsers]);
+
+    useEffect(() => {
+        // Get details of all memberships to populate dropdown
+        if(mAddresses.length > 0) {
+            setShowLoading(true);
+            const promises = mAddresses.map((mAddr) => {
+                const contract = new web3.eth.Contract(getAbiFromJson(passportJson));
+                contract.options.address = mAddr;
+                return (contract.methods.get() as ContractSendMethod).call().then((p) => {
+                    const data: MembershipDetailMap = {};
+                    data[mAddr] = {
+                        name: p[0],
+                        symbol: p[1],
+                        supply: p[2],
+                        price: web3.utils.fromWei(p[3], "ether"),
+                    };
+                    return data;
+                });
+            });
+            Promise.all(promises).then((values) => {
+                const data = Object.assign({}, ...values);
+                setMembershipDetails(data);
```

Comment:
nit: why not just do `setMembershipDetails(values)`?

Reply:
values is an array, data is an object.